src/Tree.hpp

```diff
@@ -7,12 +7,13 @@
 #include <functional>
 #include <stdexcept>
 #include <stdlib.h>
+#include <iostream>
 
-
+using namespace std;
```

Comment:
Let's remove this - it's generally bad practice to put `using ...` in headers.  [This stack overflow post](http://stackoverflow.com/questions/5849457/using-namespace-in-c-headers) has a good explanation of why.